internal/app/machined/pkg/controllers/cri/cri_test.go

```diff
@@ -0,0 +1,100 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cri_test
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/controller/runtime"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/cosi-project/runtime/pkg/state/impl/inmem"
+	"github.com/cosi-project/runtime/pkg/state/impl/namespaced"
+	"github.com/stretchr/testify/suite"
+	"github.com/talos-systems/go-retry/retry"
+
+	"github.com/talos-systems/talos/pkg/logging"
+)
+
+type CRISuite struct {
+	suite.Suite
+
+	state state.State
+
+	runtime *runtime.Runtime
+	wg      sync.WaitGroup
+
+	//nolint:containedctx
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+
+	seccompProfileDirectory string
+}
+
+func (suite *CRISuite) SetupTest() {
```

Comment:
there's new `ctest` framework by @DmitriyMV 

example: https://github.com/siderolabs/talos/blob/ae1bec59e9ce1a9b5f3a2f6a040a16bb26d991a7/internal/app/machined/pkg/controllers/secrets/kubernetes_cert_sans_test.go#L33-L42

we could use it here to give good example and also have better code

Reply:
oh that's nice, updating